checkstyle/checkstyle.xml

```diff
@@ -241,11 +241,8 @@
         <module name="JavadocMethod">
             <property name="scope" value="public"/>
             <property name="allowMissingParamTags" value="true"/>
-            <property name="allowMissingThrowsTags" value="true"/>
             <property name="allowMissingReturnTag" value="true"/>
-            <property name="minLineCount" value="2"/>
             <property name="allowedAnnotations" value="Override, Test"/>
-            <property name="allowThrowsTagsForSubclasses" value="true"/>
```

Comment:
why are these removed?

Reply:
They break checkstyle upgrade.

Reply:
just to make sure - are we making our requirements stricter by removing those rules?

Reply:
Yes, it will be stricter. I think it is okay because, currently, we do not enforce the checkstyle checks.

In the future, we should migrate to spotless, as this is what OpenSearch does: https://github.com/opensearch-project/OpenSearch/pull/1370. I will create an issue for this.